chain/builder.go

```diff
@@ -92,7 +93,7 @@ func BuildBlock(
 	}
 
 	// Compute next unit prices to use
-	feeKey := FeeKey(vm.StateManager().FeeKey())
+	feeKey := FeeKey(layout.FeePrefix())
```

Comment:
Rather than using `layout.FeePrefix()` here, which means that this is no longer configurable, could we pass a layout object into the VM, but create a default version of that layout that VMs can use, so they don't need to implement it themselves?

Then I think we can have a `layout.DefaultMetadata` or similar that we pass in to the VM constructor and use `layout.DefaultMinimumPrefix` in the VM to make sure that developers using this no exactly what prefix to start with.